lane.py

```diff
@@ -3202,9 +3202,13 @@ def configuration_relevant_update(mapper, connection, target):
     if directly_modified(target):
         site_configuration_has_changed(target)
 
-        # Some elements of Lane configuration are stored in the
-        # corresponding Library objects for performance reasons.
 
-        # Remove this information whenever the Lane configuration
-        # changes. This will force it to be recalculated.
-        Library._has_root_lane_cache.clear()
+@event.listens_for(Lane.library_id, "set")
+@event.listens_for(Lane.root_for_patron_type, "set")
+def receive_modified(target, value, oldvalue, initiator):
+    # Some elements of Lane configuration are stored in the
+    # corresponding Library objects for performance reasons.
+
+    # Remove this information whenever the Lane configuration
+    # changes. This will force it to be recalculated.
+    Library._has_root_lane_cache.clear()
```

Comment:
This fixes the failing `Lane` test. It appears that before because we were clearing the table cache during a session, we were causing items to be flushed more often and this made the "after_update" hook fire. 

It only fires after a flush to the DB. The problem with this cache was that items can be stale, without a flush happening. Hooking this set event clears this up, so that the cache is cleared when we need it to be. I'm not sure how the changes in this PR necessitated this change though.